crypto/rsa/rsa_pmeth.c

```diff
@@ -358,17 +360,29 @@ static int pkey_rsa_decrypt(EVP_PKEY_CTX *ctx,
      */
     RSA *rsa = (RSA *)EVP_PKEY_get0_RSA(ctx->pkey);
 
-    if (rctx->pad_mode == RSA_PKCS1_OAEP_PADDING) {
+    if (rctx->pad_mode == RSA_PKCS1_OAEP_PADDING
+         || rctx->pad_mode == RSA_PKCS1_WITH_TLS_PADDING) {
         if (!setup_tbuf(rctx, ctx))
             return -1;
         ret = RSA_private_decrypt(inlen, in, rctx->tbuf, rsa, RSA_NO_PADDING);
         if (ret <= 0)
             return ret;
-        ret = RSA_padding_check_PKCS1_OAEP_mgf1(out, ret, rctx->tbuf,
+
+        if (rctx->pad_mode == RSA_PKCS1_OAEP_PADDING) {
+            ret = RSA_padding_check_PKCS1_OAEP_mgf1(out, ret, rctx->tbuf,
                                                 ret, ret,
                                                 rctx->oaep_label,
                                                 rctx->oaep_labellen,
                                                 rctx->md, rctx->mgf1md);
+        } else {
+            /* RSA_PKCS1_WITH_TLS_PADDING */
+            if (rctx->client_version <= 0) {
+                ERR_raise(ERR_LIB_RSA, RSA_R_BAD_TLS_CLIENT_VERSION);
+                return 0;
+            }
+            ret = ossl_rsa_padding_check_PKCS1_type_2_TLS(NULL, out, *outlen,
+                            rctx->tbuf, ret, rctx->client_version, rctx->alt_version);
```

Comment:
nit: indentation

Reply:
Hmm, it looks like it's multiple of 4. But I guess it should be something different. I made it same way as in providers/implementation/asymciphers/rsa_enc.c . Please advice, if that is not what was expected :)

Reply:
```c
            ret = ossl_rsa_padding_check_PKCS1_type_2_TLS(NULL, out, *outlen,
                                                          rctx->tbuf, ret,
                                                          rctx->client_version,
                                                          rctx->alt_version);
```

Suggest looking at our [coding style guide](https://www.openssl.org/policies/technical/coding-style.html).